### PR TITLE
Make it easier to set-up under Docker

### PR DESCRIPTION
**Pull request by pete-woods**

### CODE REVIEW

File: src/config.cc [hunk 1]
```diff
@@ -1091,6 +1094,13 @@ std::string Config::asString() {
     return tmp;
 }
 
+void Config::docker_setup() {
+	m_logfile = std::string();
+	m_data_dir = "/tmp";
+	m_runtime_dir = "/tmp";
+	log_init(true);
+}
+
```
Comment (dwd) [inline src/config.cc]: I _think_ that if we're adjusting defaults like this based on the boot method, we should probably pass the boot method into the Config constructor and change/override from there. That'd also mean the running config was written with the right boot method, which is currently isn't. Does that make sense to you?

Comment (pete-woods) [reply src/config.cc]: Totally agree

Comment (pete-woods) [reply src/config.cc]: I thought about doing it, but didn't want to start making big changes like that to code I don't understand well, yet.

**Comment by pete-woods** [comment]
Man, that Windows build.

**Comment by pete-woods** [comment]
Apparently Docker Hub has started building PRs now?

**Pull request merged**